sqlglot/optimizer/qualify_columns.py

```diff
@@ -66,14 +66,15 @@ def qualify_columns(
             _expand_alias_refs(
                 scope,
                 resolver,
+                dialect,
                 expand_only_groupby=dialect.EXPAND_ALIAS_REFS_EARLY_ONLY_IN_GROUP_BY,
```

Comment:
Passing `dialect` here means we can refactor the `expand_only_groupby` flag as well.

Reply:
Doing that actually make some BigQuery tests fail, because this call line 77
```py
        if not schema.empty and expand_alias_refs:
            _expand_alias_refs(scope, resolver, dialect)
```

alwasy has `expand_only_groupby=False` regardless of the dialect.

Reply:
Ah, makes sense. Should we refactor this PR then to introduce a similar constant and an arg like `expand_joins`?